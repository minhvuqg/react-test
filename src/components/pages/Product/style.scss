.product-page {
  width: 100vw;

  .-container {
    display: flex;
  }
}
