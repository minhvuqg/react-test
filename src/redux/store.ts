import { combineReducers } from 'redux'
import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'

import productSlice from './reducers/productSlice'

const customizedMiddleware: any = getDefaultMiddleware({
  serializableCheck: false
})

const persistedReducer: any = combineReducers({
  cart: productSlice
})

export default configureStore({
  reducer: persistedReducer,
  middleware: customizedMiddleware
})
