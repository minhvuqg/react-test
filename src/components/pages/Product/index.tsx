import { FC } from 'react'

import ProductDetailTemplate from 'components/templates/ProductDetailTemplate'

import { IProps } from './type.d'

import './style.scss'

const Product: FC<IProps> = (): JSX.Element => {
  return (
    <div className="product-page">
      <div className="-container -card">
        <ProductDetailTemplate />
      </div>
    </div>
  )
}

export default Product
