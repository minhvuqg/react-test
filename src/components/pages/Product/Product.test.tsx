import TestRenderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import Product from '../Product'
import store from 'redux/store'
import { BrowserRouter as Router } from 'react-router-dom'

it('render product correctly', () => {
  const productEl = TestRenderer.create(
    < Provider store={store} >
      <Router>
        <Product />
      </Router>
    </Provider >).toJSON()
  expect(productEl).toMatchSnapshot()
})
