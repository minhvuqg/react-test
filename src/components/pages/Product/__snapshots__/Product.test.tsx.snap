// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render product correctly 1`] = `
<div
  className="product-page"
>
  <div
    className="-container -card"
  >
    <div>
      Not found
    </div>
  </div>
</div>
`;
